GVFS/GVFS.UnitTests/Common/GVFSEnlistmentHealthTests.cs

```diff
@@ -0,0 +1,234 @@
+﻿using GVFS.Common;
+using GVFS.Tests.Should;
+using NUnit.Framework;
+using System.Collections.Generic;
+
+namespace GVFS.UnitTests.Common
+{
+    [TestFixture]
+    public class GVFSEnlistmentHealthTests
+    {
+        private readonly char sep = GVFSPlatform.GVFSPlatformConstants.PathSeparator;
+
+        private List<string> fauxGitFiles = new List<string>();
+        private List<string> fauxGitFolders = new List<string>();
+        private List<string> fauxPlaceholderFiles = new List<string>();
+        private List<string> fauxPlaceholderFolders = new List<string>();
+        private List<string> fauxModifiedPathsFiles = new List<string>();
+        private List<string> fauxModifiedPathsFolders = new List<string>();
+        private List<string> fauxSkipWorktreeFiles = new List<string>();
+        private GVFSEnlistmentHealthCalculator enlistmentHealthCalculator;
+        private GVFSEnlistmentHealthCalculator.GVFSEnlistmentHealthData enlistmentHealthData;
+
+        [TestCase]
+        public void OnlyOneHydratedDirectory()
+        {
+            this.fauxGitFiles.AddRange(new string[] { "A/1.js", "A/2.js", "A/3.js", "A/4.js", "B/1.js", "B/2.js", "B/3.js", "B/4.js", "C/1.js", "C/2.js", "C/3.js", "C/4.js" });
+            this.fauxPlaceholderFiles.AddRange(new string[] { "A/1.js", "A/2.js", "A/3.js" });
+
+            this.enlistmentHealthData = this.GenerateStatistics(string.Empty);
+
+            this.enlistmentHealthData.DirectoryHydrationLevels[0].Key.ShouldEqual("A");
+            this.enlistmentHealthData.DirectoryHydrationLevels[0].Value.ShouldEqual(0.75m);
+            this.enlistmentHealthData.DirectoryHydrationLevels[1].Value.ShouldEqual(0);
+            this.enlistmentHealthData.DirectoryHydrationLevels[2].Value.ShouldEqual(0);
+            this.enlistmentHealthData.GitTrackedItemsCount.ShouldEqual(12);
+            this.enlistmentHealthData.PlaceholderCount.ShouldEqual(3);

```

Comment:
**NOTE: This comment applies to all unit tests**

Hardcoding what we're validating has some downsides:
- It might not be clear to the reader where the values came from
- If the test data (e.g. `fauxPlaceholderFiles`) is updated in the future all of the hardcoded values will need to be updated as well.

Rather than hardcoding these values, can we use the test data collections themselves to determine what is expected?

Example:

```
this.enlistmentHealthData.GitTrackedItemsCount.ShouldEqual(fauxGitFiles.Count);
this.enlistmentHealthData.PlaceholderCount.ShouldEqual(fauxPlaceholderFiles.Count);
```

Reply:
Tracking this in a larger GVFSEnlistmentHealhTests refactoring issue: https://github.com/microsoft/VFSForGit/issues/1299